feat: add `DistributeRewardsV2Script`

shuhuiluo opened the pull request:
**Motivation:**

https://linear.app/hnt-labs/issue/NET-157/update-llama-cron-job-proposal-to-call-notifyrewardamount-on-new

**Modifications:**

Add `DistributeRewardsV2Script` and modify interface. Run `forge fmt`.

**Result:**

Able to call `notifyRewardAmount` on new distribution contract.

shuhuiluo committed:
Add `DistributeRewardsV2Script` and update .gitignore

Introduced a new contract for distributing rewards with a specified ERC20 token and updated the interface to include a `notifyRewardAmount` function. Also, added .idea directory to .gitignore to prevent IDE metadata from being tracked.

shuhuiluo committed:
fmt

jterzis commented on src/llama-scripts/DistributeRewardsV2Script.sol:
```diff
@@ -0,0 +1,61 @@
+  IERC20 public constant RVR_TOKEN = IERC20(0x91930fd11ABAa5241241d3B07c02A8d0B5ac1920);
+
+  /// @notice The period distribution amount.
+  uint256 public constant PERIOD_DISTRIBUTION_AMOUNT = 30_769_230_769_200_000_000_000_000;
```
By hard coding this to make a change to the period amount we would need to:

1. update this script
2. redeploy the updated script
3. gov action to depermission the old script
4. gov action to permission the new script
5. pr against llama instance to update new script address

Do you see that there are multiple steps needed to make a small change to period amount ? If we read/set the period amount from governance contracts as we did previously, we would not have to undertake the above.

shuhuiluo replied:
Addressed.

shuhuiluo committed:
Refactor reward amount retrieval in reward distribution.

Replaced hardcoded reward amounts with dynamic retrieval via `getPeriodRewardAmount()` from `RegistryDiamond`. This improves flexibility and ensures the reward amount is managed directly by the registry. Added a new interface function to support this change.

shuhuiluo committed:
Add tests and improve formatting for DistributeRewardsV2

Introduce base and Sepolia tests for DistributeRewardsV2 to ensure script reliability. Additionally, simplify ERC20 transfer data formatting for improved code readability and consistency.
